Mathlib/Combinatorics/SimpleGraph/Girth.lean

```diff
@@ -9,46 +9,89 @@ import Mathlib.Data.ENat.Lattice
 /-!
 # Girth of a simple graph
 
-This file defines the girth of a simple graph as the length of its smallest cycle, or `∞` if the
-graph is acyclic.
+This file defines the girth and the extended girth of a simple graph as the length of its smallest
+cycle, they give `0` or `∞` respectively if the graph is acyclic.
 -/
 
 namespace SimpleGraph
-variable {α : Type*} {G : SimpleGraph α} {n : ℕ∞}
+variable {α : Type*} {G : SimpleGraph α}
 
-/-- The girth of a simple graph is the length of its smallest cycle, or `∞` if the graph is acyclic.
+section egirth
+
+variable {n : ℕ∞}
+
+/--
+The extended girth of a simple graph is the length of its smallest cycle, or `∞` if the graph is
+acyclic.
 -/
-noncomputable def girth (G : SimpleGraph α) : ℕ∞ :=
-⨅ a, ⨅ w : G.Walk a a, ⨅ _ : w.IsCycle, w.length
+noncomputable def egirth (G : SimpleGraph α) : ℕ∞ :=
+  ⨅ a, ⨅ w : G.Walk a a, ⨅ _ : w.IsCycle, w.length
 
-@[simp] lemma le_girth : n ≤ G.girth ↔ ∀ a (w : G.Walk a a), w.IsCycle → n ≤ w.length := by
-  simp [girth]
+@[simp] lemma le_egirth : n ≤ G.egirth ↔ ∀ a (w : G.Walk a a), w.IsCycle → n ≤ w.length := by
```

Comment:
```suggestion
@[simp] 
lemma le_egirth {n : ℕ∞} : n ≤ G.egirth ↔ ∀ a (w : G.Walk a a), w.IsCycle → n ≤ w.length := by
```

Reply:
Not sure I agree with that?

Reply:
Semantic line break triggered by line length being exceeded.

Reply:
Did I misunderstand and you are objecting to variable change?

Reply:
Ah sorry, I hadn't seen the line count